### DISCUSSION TITLE
Sign In Apple with Expo

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#2863] [Q&A]
Hey, guys,
Does anyone know how to create an authentication with apple?

there is this expo lib:
https://docs.expo.dev/versions/latest/sdk/apple-authentication/#verifying-the-response-from-apple

but I don't know if you can use it with pocketbase

**Comment by ganigeorgiev** [comment]
I personally haven't tested the Apple integration as I don't have Apple developer account, but other users reported that it works fine for them. In general you should be able to authenticate with Apple the same way as the other supported OAuth2 providers.

I don't have much experience with Expo and I'm not sure if the above component can be used (_at first glance it seems that it uses the native ios " Sign In with Apple" flow which is not supported in PocketBase_).

You can check the following Okta tutorial how to generate the required Apple keys https://developer.okta.com/blog/2019/06/04/what-the-heck-is-sign-in-with-apple#how-sign-in-with-apple-works-hint-it-uses-oauth-and-oidc (note that the Admin UI comes with secret generator, so you don't need ruby to to generate it manually as shown in the guide).

In short you'll need:
- `client_id` (aka. "Apple's Services Id")
- `client_secret` which could be generated from the Admin UI when provided:
   - `client_id`
   - `team_id`
   - `key_id`
   - `private_key` (not stored on the server; used only for the secret generation)

**Reply by brunowilliang** [reply]
I am not able to find out where I find the client_id, the other data I found
